File: backend/app/routers/health.py
from fastapi import APIRouter, Depends
from sqlalchemy import select
from sqlalchemy.ext.asyncio import AsyncSession

from app.cache import get_redis
from app.db import get_db

router = APIRouter(tags=["health"])


@router.get("/healthz")
async def healthz(db: AsyncSession = Depends(get_db)):
    status = {"status": "ok", "db": "ok", "redis": "ok"}

    try:
        await db.execute(select(1))
    except Exception as exc:  # noqa: BLE001
        status["db"] = f"error: {exc.__class__.__name__}"
        status["status"] = "degraded"

    try:
        await get_redis().ping()
    except Exception as exc:  # noqa: BLE001
        status["redis"] = f"error: {exc.__class__.__name__}"
        status["status"] = "degraded"

    return status

File: src/routes/health.js
const express = require("express");
const { sequelize } = require("../db");
const { getRedis } = require("../cache");
const { asyncHandler } = require("../middleware/asyncHandler");

const router = express.Router();

router.get("/ping", (req, res) => {
  res.status(200).json({ status: "ok" });
});

router.get(
  "/healthz",
  asyncHandler(async (req, res) => {
    const status = { status: "ok", db: "ok", redis: "ok" };

    try {
      await sequelize.query("SELECT 1");
    } catch (err) {
      status.db = `error: ${err.constructor.name}`;
      status.status = "degraded";
    }

    try {
      await getRedis().ping();
    } catch (err) {
      status.redis = `error: ${err.constructor.name}`;
      status.status = "degraded";
    }

    res.json(status);
  })
);

module.exports = router;